segregation/inference/randomization.py

```diff
@@ -0,0 +1,382 @@
+"""Tools for simulating spatial population distributions."""
+
+import itertools
+import multiprocessing
+from warnings import warn
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+from joblib import Parallel, delayed
+from tqdm.auto import tqdm
+
+
+def _generate_estimate(input):
+    if hasattr(input[0], "_original_data"):
+        df = input[0]._original_data.copy()
+    else:
+        df = input[0].data.copy()
+    if input[0].index_type == "singlegroup":
+        df = input[1](df, group=input[0].group_pop_var, total=input[0].total_pop_var,)
+        estimate = (
+            input[0]
+            .__class__(df, input[0].group_pop_var, input[0].total_pop_var, **input[2])
+            .statistic
+        )
+    else:
+        df = input[1](df, groups=input[0].groups)
+        estimate = input[0].__class__(df, input[0].groups, **input[2]).statistic
+    return estimate
+
+
+def simulate_null(
+    iterations=500,
+    sim_func=None,
+    seg_class=None,
+    n_jobs=-1,
+    backend="loky",
+    index_kwargs=None,
+):
+    """Simulate a series of index values in parallel to serve as a null distribution.
+
+    Parameters
+    ----------
+    iterations : int, optional
```

Comment:
The wording here is bit confusing. Having an optional argument followed by a required one.

Reply:
thats a typo; iterations is required